test/parallel/test-http-should-keep-alive.js

```diff
@@ -56,18 +60,17 @@ const server = net.createServer(function(socket) {
         `${SERVER_RESPONSES[responses]} should ${
           SHOULD_KEEP_ALIVE[responses] ? '' : 'not '}Keep-Alive`);
       ++responses;
-      if (responses < SHOULD_KEEP_ALIVE.length) {
+      countdown.dec();
+      if (countdown.remaining) {
         makeRequest();
-      } else {
-        server.close();
       }
       res.resume();
     });
   }
-
   makeRequest();
 });
 
+
 process.on('exit', function() {
   assert.strictEqual(requests, SERVER_RESPONSES.length);
   assert.strictEqual(responses, SHOULD_KEEP_ALIVE.length);
```

Comment:
This whole `process.on('exit')` block shouldn't be necessary if we're trying to use Countdown.

Reply:
Hi, Thanks for the quick response,
I will be happy if you can clarify some things:
1. i kept the "requests" and the "responses" variables because they are used on SERVER_RESPONSES[requests] under the makeRequest function.
since those variables are used inside of an Array if we use countdown we will get another value from the array SERVER_RESPONSES.
can you be more clear about your solution with the shift() ?

2. should I make another countdown instance for requests as well?

Reply:
There are two directions this can go in.

1. Instead of using `requests` and `responses`, just use `SERVER_RESPONSES.length -
 countdown.remaining` — that will get you the same index.
2. `shift` removes and returns the element within an array that's at index 0. You can use this to store the current value from `SERVER_RESPONSES` and `SHOULD_KEEP_ALIVE`.